spec/rubocop/cop/rails/delegate_spec.rb

```diff
@@ -249,4 +240,69 @@ def foo
       end
     RUBY
   end
+
+  it 'detects delegation to `self.class`' do
+    expect_offense(<<~RUBY)
+      def foo
+      ^^^ Use `delegate` to define delegations.
+        self.class.foo
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      delegate :foo, to: :class
+    RUBY
+  end
+
+  it 'detects delegation to a constant' do
+    expect_offense(<<~RUBY)
+      def foo
+      ^^^ Use `delegate` to define delegations.
+        CONST.foo
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      delegate :foo, to: :CONST
+    RUBY
+  end
+
+  it 'detects delegation to a namespaced constant' do
+    expect_offense(<<~RUBY)
+      def foo
+      ^^^ Use `delegate` to define delegations.
+        SomeModule::CONST.foo
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      delegate :foo, to: :'SomeModule::CONST'
+    RUBY
+  end
+
+  it 'detects delegation to an instance variable' do
+    expect_offense(<<~RUBY)
+      def foo
+      ^^^ Use `delegate` to define delegations.
+        @inst_obj.foo
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      delegate :foo, to: :@inst_obj
```

Comment:
That's my two cents. Can you tweak the example? e.g.,

```suggestion
        @instance_variable.foo
      end
    RUBY

    expect_correction(<<~RUBY)
      delegate :foo, to: :@instance_variable
```